gui/default/syncthing/core/syncthingController.js

```diff
@@ -912,6 +923,17 @@ angular.module('syncthing.core')
             return '';
         };
 
+        $scope.findDeviceFromShort = function (shortID) {
+            var matches = $scope.devices.filter(function (n) {
+                return n.deviceID.substr(0, 7) === shortID;
+            });
+            if (matches.length !== 1) {
+                n.deviceID.name = "Unknown";
+                return n;
+            }
+            return matches[0];
+        };
+
```

Comment:
Ok so you're saying "If we cant display friendly name, only then we display shortID" right?

But isn't that was it was doing before for you?  Only it didn't know the ID so it used 'AAAAAA' right?